### PR TITLE
2778 Docs copy edit

### PR DESCRIPTION
**Pull request by vikram-bitnami**
### Applicable issues

  - fixes #

### CODE REVIEW

File: dashboard/package.json [hunk 1]
```diff
@@ -17,11 +17,11 @@
     "@types/moxios": "^0.4.11",
     "@types/pako": "^1.0.1",
     "@types/qs": "^6.9.6",
-    "@types/react": "^17.0.5",
+    "@types/react": "^17.0.6",
```
Comment (vikram-bitnami) [inline dashboard/package.json:20]: Not sure why this file has come into the PR

File: dashboard/yarn.lock [hunk 2]
```diff
@@ -2283,10 +2283,10 @@
...
     "@types/react-router" "*"
 
-"@types/react-router-hash-link@^2.0.0":
-  version "2.0.0"
-  resolved "https://registry.yarnpkg.com/@types/react-router-hash-link/-/react-router-hash-link-2.0.0.tgz#93189b39dcc4ac3f28616867aad134bbe7a41521"
-  integrity sha512-Y40Kjxu9IqpVFE7gUrRJvV23+iMkrM+gI8q4DCNfywG9Cv34lQ/rrO2uZJ6TjCF2faMy3Y9rizVue1jLFQzQYA==
+"@types/react-router-hash-link@^2.4.0":
+  version "2.4.0"
+  resolved "https://registry.yarnpkg.com/@types/react-router-hash-link/-/react-router-hash-link-2.4.0.tgz#8d028a841075f06a60b502eeea251eddf39fd985"
+  integrity sha512-4IB72mosjRshMmoli6ktb0ltzEFBPWgOT/GjZwPEmcehwuacGdsTytIboPfFT9O8hgL1pgdPZMWPfMwCywjO0Q==
```
Comment (vikram-bitnami) [inline dashboard/yarn.lock:2289]: Not sure why this file has come into the PR

File: script/chart_sync_utils.sh [hunk 3]
```diff
@@ -170,11 +170,11 @@ commitAndSendExternalPR() {
     git add --all .
     git commit -m "kubeapps: bump chart version to $chartVersion"
     # NOTE: This expects to have a loaded SSH key
-    if [[ -n "$(git ls-remote origin $targetBranch  | wc -l)" ]]; then
-     echo "The remote branch '$targetBranch' already exists, please check if there is already an open PR at the repository '${CHARTS_REPO_ORIGINAL}'"
-    else
+    if [[ $(git ls-remote origin $targetBranch  | wc -l) -eq 0 ]] ; then
```
Comment (vikram-bitnami) [inline script/chart_sync_utils.sh:173]: Not sure why this file has come into the PR

File: docs/step-by-step/kubeapps-on-tkg/step-2.md [hunk 4]
```diff
@@ -35,41 +31,50 @@ kubeops: # main backend APIs configuration
...
+
+The two main configuration areas are authentication and user interface.
 
-- **Authentication** includes: i) `clusters` to define the list of clusters that Kubeapps can target and which of them will use Pinniped; ii) `pinnipedProxy`: to enable the _Pinniped Proxy_ component; iii) `authProxy`: to define the flags used by _OAuth2 Proxy_, the component for performing the actual OIDC login.
-- **Look and feel** comprises: i) `dashboard.customStyle` for injecting custom CSS; ii) `dashboard.customLocale` for customizing some supported strings; iii) `apprepository.initialRepos` for defining the repositories included by default during the installation.
+- Key authentication parameters are:
 
-> **TIP**: these values can be entered in two different ways:
+i) `clusters` to define the list of clusters that Kubeapps can target and which of them will use Pinniped;
+ii) `pinnipedProxy` to enable the _Pinniped Proxy_ component;
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-2.md:48]: It's not clear to me what rules you followed when formatting as code or wiht underscore. Can you clarify? I'm not sure why Pinniped Proxy is formatted this way.

File: docs/step-by-step/kubeapps-on-tkg/step-3.md [hunk 5]
```diff
@@ -1,138 +1,115 @@
...
-# Grant the user <USERNAME> write apprepositories permissions on the <CUSTOM-NAMESPACE> namespace
-kubectl -n <CUSTOM-NAMESPACE> create rolebinding <USERNAME>-apprepositories-write \
-    --user <USERNAME> \
+# Grant the user USERNAME write apprepositories permissions on the CUSTOM-NAMESPACE namespace
+kubectl -n CUSTOM-NAMESPACE create rolebinding USERNAME-apprepositories-write \
+    --user USERNAME \
     --clusterrole kubeapps:$KUBEAPPS_NAMESPACE:apprepositories-write
 ```
 
+TODO
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-3.md:112]: I'm not sure why this information is in this location. It does not seem related to RBAC.

File: docs/step-by-step/kubeapps-on-tkg/step-2.md [hunk 6]
```diff
@@ -203,21 +211,13 @@ roleRef:
 subjects:
   - apiGroup: rbac.authorization.k8s.io
     kind: User
-    name: email@example.com
+    name: USERNAME
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-2.md:214]: Check this

File: docs/step-by-step/kubeapps-on-tkg/step-1.md [hunk 7]
```diff
@@ -1,78 +1,95 @@
...
+Since Pinniped manages this process, the only requirement is to a _JWTAuthenticator_ CustomResource in the cluster. To do so, follow the steps below:
 
-1. Create a file `kubeapps-jwt-authenticator.yaml` with the following content:
+> **TIP**: Learn more about [JWTAuthenticator](https://pinniped.dev/docs/howto/configure-concierge-jwt/).
+
+1. Create a file named `kubeapps-jwt-authenticator.yaml` with the following content. Replace the placeholders as follows:
+
+- Replace the  `OIDC-ISSUER-URL` with the _issuer_ URL of your OIDC provider. For CSP it is `https://console-stg.cloud.vmware.com/csp/gateway/am/api`.
+- Replace `CLIENT-ID` with the application ID obtained from the JSON file in the previous step.
+- Replace `EMAIL-ADDRESS` with TODO
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-1.md:92]: Pending this information

File: docs/step-by-step/kubeapps-on-tkg/step-3.md [hunk 8]
```diff
@@ -1,138 +1,115 @@
-# Step 3 - Add Application Repositories
+## Step 3: Add Application Repositories to Kubeapps
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-3.md:2]: I feel like there is a step missing here. Shouldn't we show the user how to log in using the identity management?

File: dashboard/package.json (hunk 1 above)
Comment (antgamdia) [reply dashboard/package.json:20]: Since the branches are not in perfect sync, some changes are also included. No worries, once we merge them into my branch and, later, into the Kubeapps main branch, they will disappear.

File: dashboard/yarn.lock (hunk 2 above)
Comment (antgamdia) [reply dashboard/yarn.lock:2289]: Idem

File: docs/step-by-step/kubeapps-on-tkg/introduction.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Deploy and Configure Kubeapps on VMware Tanzu™ Kubernetes Grid
```
Comment (antgamdia) [inline docs/step-by-step/kubeapps-on-tkg/introduction.md:1]: I'm not sure, but, if we don't have any readme.md, Github won't include it in the folder previsualization. Let me explain:

https://github.com/kubeapps/kubeapps/tree/master/docs/user --> no readme, no "welcome file, no preview
https://github.com/kubeapps/kubeapps/tree/master/docs/developer --> a readme is displayed

It will depend on where do we want to put the docs, but as long as they are in gihthub, I find useful to have as a "landing" page. 

What do you think? Do you know any other alternatives?

File: script/chart_sync_utils.sh (hunk 3 above)
Comment (antgamdia) [reply script/chart_sync_utils.sh:173]: Idem

**Review by antgamdia — APPROVED** [review]

File: docs/step-by-step/kubeapps-on-tkg/step-2.md (hunk 4 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-2.md]: Any formal rule, just trying to have the most technical names with a different format (separated from `xxxxx` meaning something in the code)

File: docs/step-by-step/kubeapps-on-tkg/step-3.md (hunk 8 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-3.md:2]: Mmm, that's true. At least some screenshots with the info. I have a couple if you want them

File: docs/step-by-step/kubeapps-on-tkg/step-3.md (hunk 5 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-3.md:112]: Yes, you're right, we can safely remove it (as it is already in the kubeapps docs). I left it there because it provides more context, but in this step-by-step guide, it's unnecessary perhaps.

File: docs/step-by-step/kubeapps-on-tkg/step-3.md [hunk 10]
```diff
@@ -1,138 +1,115 @@
...
-   kibana, kiam, mariadb, postgresql-ha, grafana-operator, nginx-ingress-controller, mongodb, minio, nginx, apache, ejbca, mongodb, elasticsearch, etcd, solr, kubewatch, tomcat, kong, rabbitmq, grafana, logstash, jenkins, postgresql, redis, kube-state-metrics, airflow, mariadb-galera, contour, influxdb, cassandra, wavefront, nats, aspnet-core, wordpress, tensorflow-resnet, spring-cloud-dataflow, fluentd, zookeeper, kafka, node-exporter, memcached, redis, parse, keycloak, external-dns, pytorch, mxnet, harbor, thanos, spark, consul, kubeapps, mysql, wildfly, metrics-server
-   ```
+### Add the VMware Tanzu™ Application Catalog™ for Tanzu™ Advanced
 
-> **TIP**: To check the progress of the synchronization task, use the following command:
->
-> ```bash
-> kubectl logs -n kubeapps -l apprepositories.kubeapps.com/repo-name=tac-for-tanzu-advanced
-> ```
+The private content from the VMware Tanzu™ Application Catalog™ for Tanzu™ Advanced repository can be retrieved at `https://registry.pivotal.io/tac-for-tanzu-advanced/charts/.`. Since this is a private OCI registry, there are two key differences to note in the process:
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-3.md:56]: Please recheck dot at end of URL

File: docs/step-by-step/kubeapps-on-tkg/step-2.md (hunk 6 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-2.md:214]: AFAIK, it has to be a USER_EMAIL rather than a USERNAME (since the oidc provider just gives us the email as the user identifier). But yep, I like the capitalized approach to make it clear the things to be repaced

File: docs/step-by-step/kubeapps-on-tkg/step-2.md [hunk 11]
```diff
@@ -35,140 +31,152 @@ kubeops: # main backend APIs configuration
 
 assetsvc: # asset service configuration
 
-frontend: # nginx configuration (used to serve the dashboard and proxying the users' requests to the backend)
+frontend: # NGINX configuration (used to serve the dashboard and proxying user requests to the backend)
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-2.md:34]: We are not discussing these other parameters anywhere in this doc, should we remove them?

File: docs/step-by-step/kubeapps-on-tkg/step-3.md (hunk 10 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-3.md:56]: Oops, leftover. It should be `https://registry.pivotal.io/tac-for-tanzu-advanced/charts`
Thanks!

File: docs/step-by-step/kubeapps-on-tkg/step-2.md (hunk 11 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-2.md:34]: Yes, I guess there are not useful for the main purpose of this guide and we already have them documented in the chart, so I'm ok removing them.

File: docs/step-by-step/kubeapps-on-tkg/step-3.md [hunk 12]
```diff
@@ -1,138 +1,115 @@
...
 ```bash
-# Grant the user <USERNAME> read apprepositories permissions on the <CUSTOM-NAMESPACE> namespace
-kubectl -n <CUSTOM-NAMESPACE> create rolebinding <USERNAME>-apprepositories-read \
-    --user <USERNAME>
+# Grant the user USERNAME read apprepositories permissions on the CUSTOM-NAMESPACE namespace
+kubectl -n CUSTOM-NAMESPACE create rolebinding USERNAME-apprepositories-read \
+    --user USERNAME
     --clusterrole kubeapps:$KUBEAPPS_NAMESPACE:apprepositories-read
 
     --clusterrole kubeapps:$KUBEAPPS_NAMESPACE:apprepositories-write
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-3.md:102]: Should this line be here or is it a mistake? It is repeated again below.

File: docs/step-by-step/kubeapps-on-tkg/introduction.md (hunk 9 above)
Comment (vikram-bitnami) [reply docs/step-by-step/kubeapps-on-tkg/introduction.md:1]: Added a README with bare introduction and link. I prefer to have the actual content separated from the README as it gives more flexibility for reuse in future.

File: docs/step-by-step/kubeapps-on-tkg/step-3.md (hunk 8 above)
Comment (vikram-bitnami) [reply docs/step-by-step/kubeapps-on-tkg/step-3.md:2]: Maybe you could add that briefly with the screenshots and I can check it?

File: docs/step-by-step/kubeapps-on-tkg/step-1.md [hunk 13]
```diff
@@ -1,78 +1,91 @@
...
+Begin by creating an OAuth2 application to retrieve the information required by Pinniped and, later on, Kubeapps. Follow the steps below:
 
 > **NOTE**: You must have _Developer_ access in the organization to perform these steps.
 
-1. Go to the CSP Console available at [https://console.cloud.vmware.com](https://console.cloud.vmware.com/).
-2. Under the _Organization_ settings of the right menu, click on _View Organization_
+1. Navigate to the CSP Console at [https://console.cloud.vmware.com](https://console.cloud.vmware.com/).
+2. Click the drop-down menu in the top-right corner.
+3. Under the _Organization_ settings, click _View Organization_.
    ![View organization](./img/step-1-1.png)
```
Comment (vikram-bitnami) [inline docs/step-by-step/kubeapps-on-tkg/step-1.md:58]: There is something odd here where images are appearing in the wrong place in the github preview. Could you check why? It does not seem to happen in Step 4, but only here.

File: docs/step-by-step/kubeapps-on-tkg/step-3.md (hunk 12 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-3.md:102]: Ah, yep, we should have a code fragment for: i) apprepositories-read and ii) apprepositories-write.

It has been extracted from: https://github.com/kubeapps/kubeapps/blob/master/docs/user/access-control.md#assigning-kubeapps-user-roles

Perhaps we should point to that part directly (fine grained config) and here just create a `cluster-admin` cluster-role (all permission)

The latter is sth like:
```

# kubectl apply -f kubeapps-rbac.yaml

kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: kubeapps-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
  - apiGroup: rbac.authorization.k8s.io
    kind: User
    name: agamez@vmware.com

```

File: docs/step-by-step/kubeapps-on-tkg/step-1.md (hunk 7 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-1.md:88]: Ahh no no, it was, indeed right

```yaml
...
  claims:
    username: email
```

(This is the data which will be requested to the identity provider, that is, the usermane will be the `email` field provided by CSP)

File: docs/step-by-step/kubeapps-on-tkg/step-1.md (hunk 13 above)
Comment (antgamdia) [reply docs/step-by-step/kubeapps-on-tkg/step-1.md:58]: mmm, that's odd. I can replicate it, it's a matter of the spaces used to indent the image I think. Let me check, I'll take care of it after merging

File: docs/step-by-step/kubeapps-on-tkg/step-1.md (hunk 7 above)
Comment (vikram-bitnami) [reply docs/step-by-step/kubeapps-on-tkg/step-1.md:88]: Changed but not sure what I wrote is correct. Please check once.
